src/app/public/modules/toggle/toggle.component.spec.ts

```diff
@@ -0,0 +1,740 @@
+} from '@angular/platform-browser';
+import {
+  FormsModule,
+  NgModel,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule
+} from '@angular/forms';
+
+import {
+  expect
+} from '@skyux-sdk/testing';
+
+import {
+  SkyToggleChange,
+  SkyToggleComponent,
+  SkyToggleModule
+} from './';
+
+/** Simple component for testing a single toggle. */
+@Component({
+  template: `
+  <div>
+    <sky-toggle
+        id="simple-toggle"
+        [toggled]="isToggled"
+        [disabled]="isDisabled"
+        (change)="toggleChange($event)">
+      <sky-toggle-label>
+        Simple toggle
+      </sky-toggle-label>
+    </sky-toggle>
+  </div>`
+})
+class SingleToggleComponent {
+  public isToggled: boolean = false;
+  public isDisabled: boolean = false;
+
+  public toggleChange($event: any) {
+    this.isToggled = $event.toggled;
+  }
+}
+
+/** Simple component for testing an MdToggle with ngModel. */
+@Component({
+  template: `
+  <div>
+    <form>
+      <sky-toggle name="cb" [(ngModel)]="isGood" #wut>
+        <sky-toggle-label>
+          Be good
+        </sky-toggle-label>
+      </sky-toggle>
+    </form>
+  </div>
+  `
+})
+class ToggleWithFormDirectivesComponent {
+  public isGood: boolean = false;
+}
+
+/** Simple component for testing an MdToggle with ngModel. */
+@Component({
+  template: `
+  <div>
+    <form [formGroup]="toggleForm">
+      <sky-toggle name="tog" formControlName="toggle1" #wut>
+        <sky-toggle-label>
+          Be good
+        </sky-toggle-label>
+      </sky-toggle>
+    </form>
+  </div>
+  `
+})
+class ToggleWithReactiveFormComponent {
+  public toggle1: FormControl = new FormControl(false);
+
+  public toggleForm = new FormGroup({ 'toggle1': this.toggle1 });
+}
+
+/** Simple test component with multiple toggles. */
+@Component(({
+  template: `
+    <sky-toggle>
+      <sky-toggle-label>
+        Option 1
+      </sky-toggle-label>
+    </sky-toggle>
+    <sky-toggle>Option 2</sky-toggle>
+  `
+}))
+class MultipleTogglesComponent { }
+
+/** Simple test component with tabIndex */
+@Component({
+  template: `
+    <sky-toggle [tabindex]="customTabIndex" [disabled]="isDisabled">
+    </sky-toggle>`
+})
+class ToggleWithTabIndexComponent {
+  public customTabIndex: number = 7;
+  public isDisabled: boolean = false;
+}
+
+/** Simple test component with an aria-label set. */
+@Component({
+  template: `<sky-toggle label="Super effective"></sky-toggle>`
+})
+class ToggleWithAriaLabelComponent { }
+
+/** Simple test component with an aria-label set. */
+@Component({
+  template: `<sky-toggle labelledBy="some-id"></sky-toggle>`
+})
+class ToggleWithAriaLabelledbyComponent { }
+
+/** Simple test component with name attribute */
+@Component({
+  template: `<sky-toggle name="test-name"></sky-toggle>`
+})
+class ToggleWithNameAttributeComponent { }
+
+/** Simple test component with change event */
+@Component({
+  template: `<sky-toggle (change)="lastEvent = $event"></sky-toggle>`
+})
+class ToggleWithChangeEventComponent {
```

Comment:
I think we could combine a lot of these and then just test the functionality that we care about. I would suggest that we make a testing component and then just test the pieces of that component that we care about. I don't think we need to have separate components for toggle which only have one attribute. The test component can just have all of them and then the tests can just verify the correct things based on that.

You can also have a testing module so that you don't have to have a ton of things in the declarations for the `TestBed`

An example of this is https://github.com/blackbaud/skyux-datetime/tree/master/src/app/public/modules/date-range-picker/fixtures